lib/crypto/c_src/crypto.c

```diff
@@ -460,7 +462,9 @@ struct cipher_type_t {
 
 struct cipher_type_t cipher_types[] =
 {
+#ifndef OPENSSL_NO_RC2
```

Comment:
See the entry for `des_ede3_cfb8` a few lines below:
```
{{"des_ede3_cbf"},
#ifdef HAVE_DES_ede3_cfb_encrypt
     {&EVP_des_ede3_cfb8}
#else
     {NULL}
#endif
},
```
That is: it should set `{NULL}` when the cipher is undefined to get the right error code.